cnd/src/swap_protocols/rfc003/state_store.rs

```diff
@@ -63,192 +51,51 @@ impl StateStore for InMemoryStateStore {
             }
         };
 
-        match update {
-            SS::Start(_) => {
-                log::warn!("Attempted to update Start state for key {}", key);
-                return;
-            }
-            SS::AlphaDeployed(AlphaDeployed { alpha_deployed, .. }) => {
-                *actor_state.alpha_ledger_mut() = Deployed {
-                    htlc_location: alpha_deployed.location,
-                    deploy_transaction: alpha_deployed.transaction,
+        match event {
+            SwapEvent::AlphaDeployed(deployed) => actor_state
+                .alpha_ledger_mut()
+                .transition_to_deployed(deployed),
+            SwapEvent::AlphaFunded(funded) => {
+                let expected_asset = actor_state.expected_alpha_asset();
+
+                match expected_asset.cmp(&funded.asset) {
+                    Ordering::Equal => actor_state.alpha_ledger_mut().transition_to_funded(funded),
+                    _ => actor_state
+                        .alpha_ledger_mut()
+                        .transition_to_incorrectly_funded(funded),
                 }
             }
+            SwapEvent::AlphaRedeemed(redeemed) => {
+                // what if redeemed.secret.hash() != secret_hash in request ??
```

Comment:
If I understand correctly, it is assumed impossible because only the right secret can redeem.
So maybe worth making a note here.